### ISSUE TITLE
add readthedocs

### ISSUE THREAD

**Issue by danieleguido** [impresso/impresso-py#15]
using impresso account on readthedocs

**Comment by piconti** [comment]
I'll write some documentation [here](https://github.com/impresso/impresso-infrastructure/blob/main/howtos/how-to-update-commons.md) asap on how to start a new readthedocs. 
I will need to put my head into it again so probably will be able to have a look at it in a few days. 
I will also give @theorm or @danieleguido access to the impresso readthedocs account so that you can also go and configure what you need

**Comment by piconti** [comment]
The Impresso-py library has successfully been added to the impresso-py library. 

However, in prevision of the polar night release on May 1st, I will do a final walk-through of the docs, and see if anything is missing, with my comments here.

**Comment by theorm** [comment]
> impresso.connect - parameter public_api_url, there is ({DEFAULT_API_URL}) at the end, is it meant to be this, or should there be another value?

This is only used internally by Impresso team to connect to dev. This should not be documented and ideally should not even be visible in the documentation. It is autogenerated though and I don't think there is a way to hide it.

**Comment by theorm** [comment]
> iirc, you said that you wrote/provided some code that automatically paginates through the DataContainer and fetches the totality of the results. Is that something that is available in the library?

I don't think we ever got to doing this. Need to add a ticket for this for the next release.

**Comment by theorm** [comment]
> impresso.content_items.get() - does the CI ID need to be full? or can it be a subtring, eg. "NZZ-1794-08-09-a-i" or "NZZ-1794-08-09-a" yielding all the CIs in the issue NZZ-1794-08-09-a?

I don't know, but I suspect that's the ID of the content item in the main solr index.

**Comment by piconti** [comment]
Is the last version online? [here](https://impresso.github.io/impresso-py/result/) for example? because I'm not sure I'm seeing the modifications done to it.

Additionally, I think that @caiocmello also had additional comments on it and point for which it was unclear

**Comment by theorm** [comment]
> date_range: what is the date format? (I saw just now that this one is explained in 'preparing queries', but would be nice to have here as well)

The documentation explains it in the [type of this parameter](https://impresso.readthedocs.io/en/latest/preparing_queries/#impresso.structures.DateRange). 

> with_text_contents was confusing. I understood that TRUE means only content items with textual content. But FALSE does not mean only content items without textual content. FALSE means turn off this parameter. It took me a while to understand this. Maybe we could explain this in the documentation. Also, what are content items without textual data? Are we talking about just images? or something else?

This is an API question, which stems from how the webapp works. I leave this for @danieleguido to explain.

> collection_id: this is a comment about all the id parameters. I think we should explain how you find those ids. It's quite difficult to understand where I find the id that I can use to make searches using this.

Can you please update the documentation explaining that it can be found by listing all collections with `collections` resource.

> For some, like partner_id, could we have a list of partners ids?

The list can be obtained by inspecting the partner facet.
